### PR TITLE
Merge controller subcommand into connect inject subcommand

### PR DESCRIPTION
**Pull request by thisisnotashwin**
Changes proposed in this PR:
- Merge the controller subcommand into the connect inject subcommand.
- This requires the CRDs to be deployed into the cluster when connectInject.enabled = true.
- The ACL policy for connect inject needs to be updated to be a super set and include controller policies where it previously didnt.
- Delete the controller sections from the values files, webhook config, the acceptance tests and the project at large.

How I've tested this PR:
- Acceptance and unit tests

How I expect reviewers to test this PR:
- 👀 
- Anything missing?

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

**Review by curtbushko — COMMENTED** [review]
Looks Good!  There are a lot of changes but the tests make this feel a lot safer to do.

I only had a couple questions but nothing I would block you on. Ping me tomorrow if you need an approval.

File: charts/consul/values.yaml [hunk 1]
```diff
@@ -2302,82 +2302,6 @@ connectInject:
...
-controller:
-  # Enables the controller for managing custom resources.
-  enabled: true
-
-  # The number of deployment replicas.
-  replicas: 1
-
-  # Log verbosity level. One of "debug", "info", "warn", or "error".
-  # @type: string
-  logLevel: ""
```
[the remark this answers is not in the document]
Comment (thisisnotashwin) [reply charts/consul/values.yaml:2317]: Those options are identical to those that already exist under connectInject and we will just inherit those values. There does exist a `connectInject.logLevel` specifically though 😄

[the remark this answers is not in the document]
Comment (curtbushko) [reply charts/consul/values.yaml:2317]: Oh darn it!  The connectInject section is so big that I missed it!

**Review by ishustava — APPROVED** [review]
Awesome work!! Please remember to add a changelog entry with the breaking change

**Pull request merged**